src/System.Windows.Extensions/src/System/Drawing/FontConverter.cs

```diff
@@ -126,94 +123,179 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
+            string f_name = font; // start with the assumption that only the font name was provided.
+            string styleStr = null;
+            string sizeStr = null;
+            float f_size = 8.25f;
+            FontStyle f_style = FontStyle.Regular;
+            GraphicsUnit units = GraphicsUnit.Point;
+
+            // Get the index of the first separator (would indicate the end of the name in the string).
+            int nameIndex = font.IndexOf(sep);
+
+            if (nameIndex < 0)
             {
-                throw new ArgumentException("Failed to parse font format");
+                return new Font(f_name, f_size, f_style, units);
             }
 
-            font = fields[0];
-            f_size = 8f;
-            units = "px";
-            f_unit = GraphicsUnit.Pixel;
-            if (fields.Length > 1)
-            {   // We have a size
-                for (int i = 0; i < fields[1].Length; i++)
+            // some parameters are provided in addition to name.
+            f_name = font.Substring(0, nameIndex);
+
+            if (nameIndex < font.Length - 1)
+            {
+                // Get the style index (if any). The size is a bit problematic because it can be formatted differently 
+                // depending on the culture, we'll parse it last.
+                int styleIndex = font.IndexOf(styleHdr, StringComparison.CurrentCultureIgnoreCase);
+
+                if (styleIndex != -1)
                 {
-                    if (char.IsLetter(fields[1][i]))
+                    // style found.
+                    styleStr = font.Substring(styleIndex, font.Length - styleIndex);
+
+                    // Expected style format ~ "style=Italic, Bold"
+                    if (!styleStr.StartsWith(styleHdr, StringComparison.CurrentCultureIgnoreCase))
                     {
-                        f_size = (float)TypeDescriptor.GetConverter(typeof(float)).ConvertFromString(context, culture, fields[1].Substring(0, i));
-                        units = fields[1].Substring(i);
-                        break;
+                        throw new ArgumentException(SR.Format(SR.TextParseFailedFormat, font, $"name{sep} size[units[{sep} style=style1[{sep} style2{sep} ...]]]"));
                     }
+
+                    // Get the mid-substring containing the size information.
+                    sizeStr = font.Substring(nameIndex + 1, styleIndex - nameIndex - 1);
                 }
-                switch (units)
+                else
                 {
-                    case "display":
-                        f_unit = GraphicsUnit.Display;
-                        break;
+                    // no style.
+                    sizeStr = font.Substring(nameIndex + 1);
+                }
 
-                    case "doc":
-                        f_unit = GraphicsUnit.Document;
-                        break;
+                // Parse size.
+                (string, string) unitTokens = ParseSizeTokens(sizeStr, sep);
```

Comment:
Can we use named value tuples? Will make it easier to read, instead of using `Item1,Item2,etc`.